src/simplify_algebra.cpp

```diff
@@ -1200,6 +1200,12 @@ struct find_div_const
         auto ins   = r.result;
         auto c_ins = r.instructions["c"];
 
+        bool is_integral = false;
+        c_ins->get_shape().visit_type([&](auto t) { is_integral = t.is_integral(); });
+        // avoid 1 / integer
+        if(is_integral)
+            return;
```

Comment:
A similiar fix was added in #2547. We should merge #2547 in first before merging in this PR:

https://github.com/ROCm/AMDMIGraphX/pull/2547/files#diff-0e2e45cd7b8d50c1e0790164b152eb4c8ef7c58c3baffd88e3bed1cb2a754d8bR1203